src/fields/Url.php

```diff
@@ -95,29 +138,167 @@ public function getSettingsHtml()
             ]);
     }
 
+    /**
+     * @inheritdoc
+     */
+    public function normalizeValue($value, ElementInterface $element = null)
+    {
+        if (is_array($value) && isset($value['type'])) {
+            $type = $value['type'];
+            $value = trim($value[$type]);
+
+            if ($value) {
+                switch ($type) {
+                    case self::TYPE_TEL:
+                        $value = str_replace(' ', '-', $value);
+                        $value = StringHelper::ensureLeft($value, 'tel:');
+                        break;
+                    case self::TYPE_EMAIL:
+                        $value = StringHelper::ensureLeft($value, 'mailto:');
+                        break;
+                    case self::TYPE_URL:
+                        if (!UrlHelper::isFullUrl($value)) {
+                            $value = StringHelper::ensureLeft($value, 'http://');
```

Comment:
Not sure, but would this be better off using `https://` or `//` instead of `http://`?
https://www.paulirish.com/2010/the-protocol-relative-url/

Reply:
Most sites that support SSL will automatically redirect to it, whereas there are plenty of sites that _only_ support `http` still, so I figured `http` is a better default.